### ISSUE TITLE
Playlist timeline polish / improvements / QOL

### ISSUE THREAD

**Issue by tetsuoanimation** [cgwire/kitsu#1366]
_Note: This issue is located somewhere between a bug and feature request - feel free to move it to the category you want to sue for it_

**Context**

Studio name: Tetsuo Animation
Kitsu version: 0.19.2
Production type: Short

**Description**
- Dragging on the timeline should be locked to the timeline the drag was started on:
  - When starting to drag on the frame timeline, only change the frames in the player until drag/touch is released  ( even if hovering over the main timeline )
  - When starting to drag on the overall timeline, only change the frames in the main timeline until drag/touch is released ( even if hovering over the frame timeline )
- Preview images cover the control/playback buttons and don't go away while hovering over the text on them - this hides the play-button and throws off the flow
- Hovering over the main timeline is lost when the timeline div animates away under the mouse cursor.
  - Leads to flickering between the preview images
  - Means that the clip jumps between the main and frame timeline frames/clips when clicked.
  - Preview images might not be needed on one of the timelines maybe - other users might differ.
- Clip looping is not instant
- Clicking the timeline to drag gets a little annoying after a while. Mirroring what Maya does (dragging on the viewport while holding 'K' is scrubbing the timeline) or click-dragging on the video viewport for scrubbing might be helpful (this needs a quick shortcut to switch between selecting, drawing and scrubbing then)

**Comment by frankrousseau** [comment]
> -    When starting to drag on the frame timeline, only change the frames in the player until drag/touch is released ( even  if hovering over the main timeline )
> -    When starting to drag on the overall timeline, only change the frames in the main timeline until drag/touch is released ( even if hovering over the frame timeline )

HD videos are heavy but it should works as expected with LD videos.

**Comment by frankrousseau** [comment]
For caching ideas: https://web.dev/articles/fast-playback-with-preload

**Comment by frankrousseau** [comment]
I add some shortcuts requested by a French customer:

En gros on a acces à fleche gauche, fleche droite et ALT-J  et ALT-K.
J'imagine que la touche HOME pourrait nous amener en début de plan, la touche END en fin.
Et ALT-fleche gauche/ALT-fleche droite  se déplacer sur la timeline (pouvoir sauter par delà les cuts)

**Comment by PeteDraper** [comment]
I'd also like to throw watermarking, company logo (can be taken from uploaded company image from Kitsu) and custom text along with tags for artist name, project, sequence, shot, tc etc. attached is a python script I use for watermarking confidential videos - it resizes to 1920x1080, adds the logo in the top left, caption across the bottom and can batch a folder. Hope it helps 
[Archive.zip](https://github.com/cgwire/kitsu/files/15324671/Archive.zip)

**Comment by tetsuoanimation** [comment]
@PeteDraper I do this with ffmpeg currently. Here's my repo: https://github.com/tetsuoanimation/editbot
This is abit messy ( it was a backend oneshot ) but it could help you with the features you're requesting.

**Comment by PeteDraper** [comment]
this is great - would be good to have something like this in Kitsu when downloading the playlist build

**Comment by tetsuoanimation** [comment]
> I add some shortcuts requested by a French customer:
> 
> En gros on a acces à fleche gauche, fleche droite et ALT-J et ALT-K. J'imagine que la touche HOME pourrait nous amener en début de plan, la touche END en fin. Et ALT-fleche gauche/ALT-fleche droite se déplacer sur la timeline (pouvoir sauter par delà les cuts)

These are great! I didn't know them and I'm sure I'll use them a bunch!
I'd love it if there was a shortcut to jump between drawings ( , . like in maya wouble be awesome! )

**Comment by frankrousseau** [comment]
FYI, most problems are fixed but I keep the issue open because of the watermarking and caching discussions.

**Comment by tetsuoanimation** [comment]
@frankrousseau Regarding this discussion - are the 'other' shortcuts ( jumping to the next / previous drawing, hold key and drag to scrub and switching focus between drawing / playback and the comment / notebox ) still on the table?

I also noticed that the 'internal' variable for the frame is not updated when using the shortcuts you mentioned to go to the next or previous shot ( alt+j/k or alt+arrowkeys ). While the timeline shows the first frame, the system remembers where the pleayhead was in the previous shot when you press the arrow keys to advance a single frame.

**Comment by frankrousseau** [comment]
Yes, it's still interesting, but we are swamped with the many feature requests + vue 3 migration. Vue 3 migration is close to be finished, so I hope we will be able to get faster with the maintenance after that.

**Comment by frankrousseau** [comment]
Most of the issues have been fixed. Please reopen an issue if needed (watermaking is not developed).
